src/DiskIO/DiskThreads/CommIO.cc

```diff
@@ -10,6 +10,7 @@
 
 #include "squid.h"
 #include "comm/Loops.h"
+#include "compat/pipe.h"
```

Comment:
If you are going to insist on having a whole header file just for `pipe()` definition. It should be included into the `compat/os/mingw.h` and `compat/os/mswindows.h` header files.
At worst, it could go at the end of `compat/compat.h` after the `debug.h` include line.

If you are doing this because of issues the `.c` files have - then wrap this `pipe()` definition with:
```c++
#if defined(__cplusplus)
```